framework/auth/views.py

```diff
@@ -193,23 +193,40 @@ def confirm_email_get(**kwargs):
     methods: GET
     """
     user = User.load(kwargs['uid'])
+    is_initial_confirmation = not user.date_confirmed
     token = kwargs['token']
-    if user:
-        if user.confirm_email(token):  # Confirm and register the user
-            user.date_last_login = datetime.datetime.utcnow()
-            user.save()
 
-            # Go to settings page
-            status.push_status_message(language.WELCOME_MESSAGE, 'success')
-            response = redirect('/settings/')
+    if user is None:
+        raise HTTPError(http.NOT_FOUND)
 
-            return framework.auth.authenticate(user, response=response)
-    # Return data for the error template
-    return {
-        'code': http.BAD_REQUEST,
-        'message_short': 'Link Expired',
-        'message_long': language.LINK_EXPIRED
-    }, http.BAD_REQUEST
+    try:
+        user.confirm_email(token)
+    except exceptions.EmailConfirmTokenError as e:
+        return {
```

Comment:
I agree. I was following the convention of the existing code so as not to have to think about the impact of refactoring it. There's shouldn't be an impact to this change though, so I've made it.